### PR TITLE
Abel presubmit workflow

### PR DESCRIPTION
**Pull request by trav3711**

### CODE REVIEW

File: .github/workflows/tools.yml [hunk 1]
```diff
@@ -7,105 +7,133 @@ on:
...
 jobs:
   build:
+    name: [ master-runner ]
     runs-on: ${{ matrix.os }}
     timeout-minutes: 60
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest, windows-latest]
     steps:
     - uses: actions/checkout@v2
```
Comment (charbull) [inline .github/workflows/tools.yml:25]: remove this?

File: .github/workflows/tools.yml [hunk 2]
```diff
@@ -7,105 +7,133 @@ on:
...
         python -m pip install pylint
+    - name: Run Pylint on tools
+      run: |
+        pylint --rcfile ./styles/pylintrc ./tools
+
 
      #TODO(b/210730979): Investigate calling ontology validator setup with instance validator setup
      #---------- Ontology Explorer ----------#
-    - name: Ontology Validator Install dependencies
-      run: |
```
Comment (charbull) [inline .github/workflows/tools.yml:36]: why deleting those?

File: .github/workflows/tools.yml [hunk 3]
```diff
@@ -7,105 +7,133 @@ on:
...
-        python -m pip install .
-      working-directory: ./tools/validators/instance_validator
     - name: Ontology Explorer Install dependencies
+      if: steps.changes.outputs.explorer == 'true'
       run: |
         python setup.py install
       working-directory: ./tools/explorer
-    - name: Run Pylint on Ontology Explorer
-      run: |
```
Comment (charbull) [inline .github/workflows/tools.yml:48]: pylints must stay?

File: .github/workflows/tools.yml [hunk 4]
```diff
@@ -7,105 +7,133 @@ on:
...
-        pylint --rcfile ./styles/pylintrc ./tools/explorer
     - name: Ontology Explorer Tests
```
Comment (charbull) [inline .github/workflows/tools.yml:63]: this will break though? it will not find the ontology type validator dependencies

Comment (trav3711) [reply .github/workflows/tools.yml:63]: I fixed the order of operations.

File: .github/workflows/tools.yml (hunk 2 above)
Comment (trav3711) [reply .github/workflows/tools.yml:36]: I fixe the order of operations, so each tool installs only its own library.

File: .github/workflows/tools.yml (hunk 1 above)
Comment (trav3711) [reply .github/workflows/tools.yml:25]: weird....I did't add this. Removed.

File: .github/workflows/tools.yml [hunk 5]
```diff
@@ -31,81 +49,114 @@ jobs:
...
+      if: steps.changes.outputs.explorer == 'true'
+      run: |
+        pylint --rcfile ./styles/pylintrc ./tools/explorer
+    - name: Ontology Explorer Tests
+      if: steps.changes.outputs.explorer == 'true'
+      run: |
+       python -m unittest discover -p '*_test.py' --failfast -v
+      working-directory: ./tools/explorer
+
     #---------- Configuration scoring ----------#
```
Comment (charbull) [inline .github/workflows/tools.yml:99]: nit move this last after the rdf generator it is the least that is going to get updated

File: .github/workflows/tools.yml [hunk 6]
```diff
@@ -31,81 +49,114 @@ jobs:
...
     - name: Configuration scoring Install dependencies
+      if: steps.changes.outputs.scoring == 'true'
       run: |
         python setup.py install
       working-directory: ./tools/scoring
-    - name: Run Pylint on Configuration Scorer
+    - name: Run pylint on Instance Scorer
```
Comment (charbull) [inline .github/workflows/tools.yml:105]: nit: keep the same name please: Configuration Scorer

File: .github/workflows/tools.yml [hunk 7]
```diff
@@ -31,81 +49,114 @@ jobs:
...
+      run: |
+        pylint --rcfile ./styles/pylintrc ./tools/rdf_generator
     - name:  RDF Generator Tests
+      if: steps.changes.outputs.rdf_generator == 'true'
       run: |
        python -m unittest discover -p '*_test.py' --failfast -v
       working-directory: ./tools/rdf_generator/rdfformat/tests
-    - name: Run Pylint on RDF Generator
+
+      #---------- ABEL ----------#
```
Comment (charbull) [inline .github/workflows/tools.yml:148]: nit move abel above RDF generator?

**Review by charbull — CHANGES_REQUESTED** [review]

File: .github/workflows/tools.yml (hunk 6 above)
Comment (trav3711) [reply .github/workflows/tools.yml:105]: Done.

File: .github/workflows/tools.yml (hunk 7 above)
Comment (trav3711) [reply .github/workflows/tools.yml]: Done.

File: .github/workflows/tools.yml (hunk 5 above)
Comment (trav3711) [reply .github/workflows/tools.yml]: Done. Moved scorer to last.

**Comment by trav3711** [comment]
Completely removed ABEL from the workflow because tests were also having trouble.

**Review by charbull — APPROVED** [review]

**Pull request merged**